### ISSUE TITLE
How to update DATABASE when updating the version of lightningd?

### ISSUE THREAD

**Issue by toshirot** [ElementsProject/lightning#202]
Hi, i Hope you are doing well.
I am looking forward to your work!

Today I did update the lightningd.

--version
v0.5.2-2016-11-21-731-gfbe3a01

From

--version
v0.5.2-2016-11-21-727-g3584720

After that, when i try to start lightningd,
It is said.

DATABASE NEEDS UPDATE. Version v0.5.2-2016-11-21-727-g3584720 does not match v0.5.2-2016-11-21-731-gfbe3a01

How can I update the database for the new version.

Do I have to make a new --lightning-dir ?

**Comment by cdecker** [comment]
Hi @toshirot, thanks for the report. From the error I assume you are using the legacy version of `daemon/lightningd` which is going to be retired soon. The new (and supported) version is `lightningd/lightningd` which includes an automatic update mechanism for the database. If you don't have channels that you need to keep around I'd say delete the database at `$HOME/.lightning/lightning.sqlite3`. If you would like to simply start without checking the DB compatibility there's the `--ignore-dbversion`, it's very likely that the DB schema did not change and it'll just run, though I can't guarantee it :wink:

**Comment by toshirot** [comment]
Thank you for the advice :)

**Comment by toshirot** [comment]
oh! i want to install the new (and supported) version.
is it v0.6?
What should I do now?

**Comment by cdecker** [comment]
We are currently working on getting the new daemon (which eventually will be released as v0.6) feature complete, so there currently is no release. However you can grab the sources and compile them yourself if you'd like to help testing. The [readme](https://github.com/ElementsProject/lightning#installation) has the minimal instructions to compile from source, but feel free to contact me for help :wink:

**Comment by toshirot** [comment]
Thank you!
i tried follows code. https://github.com/ElementsProject/lightning#installation

--
cd ~
mv lightning lightning-old-0810 #backup

sudo apt-get install -y autoconf git build-essential libtool libprotobuf-c-dev libgmp-dev libsqlite3-dev python python3
git clone https://github.com/ElementsProject/lightning.git
cd lightning
make

--

But version is displayed as v 0.5.2

./daemon/lightning-cli --version
v0.5.2-2016-11-21-752-gfc59d8e

**Comment by cdecker** [comment]
That is correct, the version is bumped as the last step before packaging the release. The version string shows that you are running the latest commit fc59d8e227ca6a4b30902698a9355c5aebddba6e. However instead of starting `daemon/lightningd` you should start `lightningd/lightningd`. I know this is confusing, we will retire/remove the old daemon soon.

**Comment by toshirot** [comment]
Thank you:) There was lightningd/lightningd!

I tested follows code.


    ----------------------------------------
    mkdir for channels
    
    tato@hoge:~/lightning$ ls -l ~/lgt
    合計 28
    drwxrwxr-x 2 tato tato 4096  8月 11 08:18 test10
    drwxrwxr-x 2 tato tato 4096  8月 11 08:16 test11
    
    ----------------------------------------
    run test10
    
    tato@hoge:~/lightning$ lightningd/lightningd --port 42010 --lightning-dir /home/tato/lgt/test10 --log-file /home/tato/lgt/test10/log.txt &
    
    lightningd(12287): Hello world from v0.5.2-2016-11-21-752-gfc59d8e!
    lightningd(12287): Our ID: 02a16379047b7cdd8db105be60d0ec58ba27b4845f2b138d85a0b4a1b28a52ca98
    lightningd_gossip(12296): pid 12296, msgfd 13
    lightningd(12287): Failed to bind on 2 socket: Address already in use
    
    ----------------------------------------
    run test11
    
    tato@hoge:~/lightning$ lightningd/lightningd --port 42011 --lightning-dir /home/tato/lgt/test11 --log-file /home/tato/lgt/test11/log.txt &
    
    lightningd(11302): Hello world from v0.5.2-2016-11-21-752-gfc59d8e!
    lightningd(11302): Our ID: 03706392fa1fec281f92087885c935b7e66316b6d18ae5ee459564beb03ff64631
    lightningd_gossip(11311): pid 11311, msgfd 13
    lightningd(11302): Failed to bind on 2 socket: Address already in use
    
    ----------------------------------------
    getinfo
    
    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test10 getinfo 
    { "id" : "02a16379047b7cdd8db105be60d0ec58ba27b4845f2b138d85a0b4a1b28a52ca98", "port" : 42010, "testnet" : true, "version" : "v0.5.2-2016-11-21-752-gfc59d8e", "blockheight" : 1157133 }
    
    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test11 getinfo 
    { "id" : "03706392fa1fec281f92087885c935b7e66316b6d18ae5ee459564beb03ff64631", "port" : 42011, "testnet" : true, "version" : "v0.5.2-2016-11-21-752-gfc59d8e", "blockheight" : 1157135 }
    
    ----------------------------------------
    mk address

    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test10 newaddr
    { "address" : "2MxwMM7ACVox2QkMZ5BfMv71bWVNHjmoEoG" }
    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test11 newaddr
    { "address" : "2NDvUWQrX7a7jxL2KdR1dX4sPVmXmEKqbyE" }
    
    ----------------------------------------
    check the testnet btc by bitcoin-cli
    
    tato@hoge:~/lightning$ bitcoin-cli -testnet getbalance
    0.37734710
    
    ----------------------------------------
    send btc by bitcoin-cli sendtoaddress (@return txid)

    tato@hoge:~/lightning$ bitcoin-cli -testnet sendtoaddress 2MxwMM7ACVox2QkMZ5BfMv71bWVNHjmoEoG 0.003
    4fc008a8fc1900ca34edad144dd6d87b474f80a56b26f9fb7477ef3698972236
    tato@hoge:~/lightning$ bitcoin-cli -testnet sendtoaddress 2NDvUWQrX7a7jxL2KdR1dX4sPVmXmEKqbyE 0.004
    4c877bcbdc674bd48067dcc75e93d598d4f44aedc1cece504514faf49353c8c2
    
    
    ----------------------------------------
    getrawtransaction txid (@return rawtx)
    
    //test10
    tato@hoge:~/lightning$ bitcoin-cli -testnet getrawtransaction 4fc008a8fc1900ca34edad144dd6d87b474f80a56b26f9fb7477ef3698972236
    02000000011f27eefde8668ff2a29f25092eafb8781f545552ef01ebc19dfadf70ed7e0d4a010000006b483045022100cf15482511cbe770247df0be81318408bf49bacee36883007e28735f313bcd8e022014d61cb21f2fdacc63db222b7aca6d958859f2fa85425da5cdacc7e2597972a1012102d13e6f70a84c5af9b06f8f3111cc6b480b7d304aa1e71ba91f521049e2c9f36ffeffffff02e09304000000000017a9143e6f690f4713d8003bb2e3bae0b64b046587e19787f1311c00000000001976a914d58ea22f4e96d111f92d94f61bfdf2363a7c7f5588ac10a81100
    
    //test11
    tato@hoge:~/lightning$ bitcoin-cli -testnet getrawtransaction 4c877bcbdc674bd48067dcc75e93d598d4f44aedc1cece504514faf49353c8c2
    0200000001e56cc9efadaddb3dbf81a715e229c8a413f26ff4ee9e9ab603a48978a9562f24000000006a47304402203f0f01d1819b860dd7a9c9c446d0c513c9bb4fa6a2b6af22b1810b70112709e902202a74e8cfac1d84128aec0e1687dc2b98dc4e6d1f5257019a3e980f6be38ef61c0121034817a0433bfa71090e2d89adbe2d57a939929b428849bc5db154fc8dd293804afeffffff02801a06000000000017a914e2cedbc0e95f93a6abe7d50da91ee8b1497312d08767e51802000000001976a914fd02098a17ed2216af2b99ffaf88421aa1f4619288ac10a81100
    
    
    ----------------------------------------
    addfunds rawtx

    //test10
    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test10 addfunds 02000000011f27eefde8668ff2a29f25092eafb8781f545552ef01ebc19dfadf70ed7e0d4a010000006b483045022100cf15482511cbe770247df0be81318408bf49bacee36883007e28735f313bcd8e022014d61cb21f2fdacc63db222b7aca6d958859f2fa85425da5cdacc7e2597972a1012102d13e6f70a84c5af9b06f8f3111cc6b480b7d304aa1e71ba91f521049e2c9f36ffeffffff02e09304000000000017a9143e6f690f4713d8003bb2e3bae0b64b046587e19787f1311c00000000001976a914d58ea22f4e96d111f92d94f61bfdf2363a7c7f5588ac10a81100                                                                            
    { "outputs" : 1, "satoshis" : 300000 }

    //test11
    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test11 addfunds 0200000001e56cc9efadaddb3dbf81a715e229c8a413f26ff4ee9e9ab603a48978a9562f24000000006a47304402203f0f01d1819b860dd7a9c9c446d0c513c9bb4fa6a2b6af22b1810b70112709e902202a74e8cfac1d84128aec0e1687dc2b98dc4e6d1f5257019a3e980f6be38ef61c0121034817a0433bfa71090e2d89adbe2d57a939929b428849bc5db154fc8dd293804afeffffff02801a06000000000017a914e2cedbc0e95f93a6abe7d50da91ee8b1497312d08767e51802000000001976a914fd02098a17ed2216af2b99ffaf88421aa1f4619288ac10a81100
    { "outputs" : 1, "satoshis" : 400000 }
    
    ----------------------------------------
    connect test10 to test11

    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test10 connect 192.168.1.10 42011 0200000001e56cc9efadaddb3dbf81a715e229c8a413f26ff4ee9e9ab603a48978a9562f24000000006a47304402203f0f01d1819b860dd7a9c9c446d0c513c9bb4fa6a2b6af22b1810b70112709e902202a74e8cfac1d84128aec0e1687dc2b98dc4e6d1f5257019a3e980f6be38ef61c0121034817a0433bfa71090e2d89adbe2d57a939929b428849bc5db154fc8dd293804afeffffff02801a06000000000017a914e2cedbc0e95f93a6abe7d50da91ee8b1497312d08767e51802000000001976a914fd02098a17ed2216af2b99ffaf88421aa1f4619288ac10a81100
    
    "id 0200000001e56cc9efadaddb3dbf81a715e229c8a413f26ff4ee9e9ab603a48978a9562f24000000006a47304402203f0f01d1819b860dd7a9c9c446d0c513c9bb4fa6a2b6af22b1810b70112709e902202a74e8cfac1d84128aec0e1687dc2b98dc4e6d1f5257019a3e980f6be38ef61c0121034817a0433bfa71090e2d89adbe2d57a939929b428849bc5db154fc8dd293804afeffffff02801a06000000000017a914e2cedbc0e95f93a6abe7d50da91ee8b1497312d08767e51802000000001976a914fd02098a17ed2216af2b99ffaf88421aa1f4619288ac10a81100 not valid"

oops!
connect syntax is 

``
  daemon/lightning-cli connect <ip> <port> <node_id>
``

I think in this case

``
  daemon/lightning-cli  --lightning-dir <test10-dir>  connect <test11-ip> <test11-port> <test11-node_id>
``

<test11-node_id> is not rawtx ?

**Comment by cdecker** [comment]
That's great progress @toshirot, the `<node_id>` refers to the `id`-field returned from `getinfo`. The `rawtx` is no longer needed since the new wallet includes its own wallet and tracks its outputs, and opening a channel is now a 2 step process:

```
# Open a connection from test10 to test11
./daemon/lightning-cli --lightning-dir /home/tato/lgt/test10 connect 192.168.1.10 42011 03706392fa1fec281f92087885c935b7e66316b6d18ae5ee459564beb03ff64631

# Use the connection to open a channel
./daemon/lightning-cli --lightning-dir /home/tato/lgt/test10 fundchannel 03706392fa1fec281f92087885c935b7e66316b6d18ae5ee459564beb03ff64631 200000
```
That should open a connection and establish a channel. Then wait for 2 confirmations and your channel is operational. Finally, there is no real need to add funds to node `test11`, since only funds from `test10`, i.e., the channel funder, are used in the establishment of the channel.

**Comment by toshirot** [comment]
Great!
and, I am very thankful for your support :)

Well, I tested the continuation.
This time i used test0 and test1 directories.



    ----------------------------------------
    getinfo

    //test0
    tato@finchat:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test0 getinfo
    { "id" : "031824442e75ecf3ce0a535a5cb2b2262e3fea76b01ff2e1bf8687e00ba32b8c8e", "port" : 42000, "testnet" : true, "version" : "v0.5.2-2016-11-21-752-gfc59d8e", "blockheight" : 1163801 }

    //test1
    tato@finchat:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test1 getinfo
    { "id" : "02cd23430288d269b9383895272f48f0b8489bba1cca67ecca1833a32850d1ce0b", "port" : 42001, "testnet" : true, "version" : "v0.5.2-2016-11-21-752-gfc59d8e", "blockheight" : 1163806 }

    ----------------------------------------
    Open a connection from test0 to test1

    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test0 connect 192.168.1.10 42001 02cd23430288d269b9383895272f48f0b8489bba1cca67ecca1833a32850d1ce0b
    { "id" : "02cd23430288d269b9383895272f48f0b8489bba1cca67ecca1833a32850d1ce0b" }


    ----------------------------------------
    Use the connection to open a channel

    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test0  fundchannel 02cd23430288d269b9383895272f48f0b8489bba1cca67ecca1833a32850d1ce0b 2000
    {  }

    ----------------------------------------
    creates an invoice

    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test1 invoice 1000 tx-0812-01
    { "rhash" : "6a6e0f0b3491bf4d52fd5d87a101c58a1951d8232381a093c5827449873c72af" }

    ----------------------------------------
    compute a route

    tato@hoge:~/lightning$ route=$(./daemon/lightning-cli --lightning-dir /home/tato/lgt/test0  getroute 02cd23430288d269b9383895272f48f0b8489bba1cca67ecca1833a32850d1ce0b 1000 1 | jq --raw-output .route -)


    tato@hoge:~/lightning$ echo $route
    [ { "id": "02cd23430288d269b9383895272f48f0b8489bba1cca67ecca1833a32850d1ce0b", "channel": "1163841:1:0", "msatoshi": 
    , "delay": 4 } ]


    ----------------------------------------
    sendpay from test0 to test1

    tato@hoge:~/lightning$ ./daemon/lightning-cli --lightning-dir /home/tato/lgt/test0  sendpay $route 1000
    lightning-cli: Non-object response ''




Oh orz..

According to
https://github.com/ElementsProject/lightning#user-content-receiving-and-receiving-payments
sendpay syntax is

``
daemon/lightning-cli sendpay $route <amount>
``

But, according to
https://github.com/ElementsProject/lightning/blob/master/doc/lightning-sendpay.7.txt

``
*sendpay* 'route' 'hash'
``

not amount?

**Comment by cdecker** [comment]
Great catch @toshirot, the manpage is correct in this case, since the amount is already encoded in the route. I fixed the documentation to reflect this f802e98e73bc4e42dd70e58069652703ce6094e4

**Comment by jl777** [comment]
use the rhash value 6a6e0f0b3491bf4d52fd5d87a101c58a1951d8232381a093c5827449873c72af instead of <amount>

**Comment by jl777** [comment]
it looks like the latest version automatically adds all utxo with an automatic addfunds?
is there a way to get a list of all the added utxo and/or the current balance remaining

**Comment by cdecker** [comment]
@jl777 could you open a new issue for this? But basically it'll tell you how many outputs it has detected and what the total detected value is.

**Comment by toshirot** [comment]
Today I did the next test
It may be a bit more:?



    mkdir -p ~/lgt/test12
    mkdir -p ~/lgt/test13

    #================================
    # Run  localhost:42012 test12 


      #--------------------------------
      # Start lightningd  test12 

        $ $HOME/lightning/lightningd/lightningd  --port 42012 --lightning-dir /home/tato/lgt/test12 --log-file /home/tato/lgt/test12/log.txt & 


      #--------------------------------
      # tail 5 lines of logs: /home/tato/lgt/test12/log.txt 

        $ tail -5 /home/tato/lgt/test12/log.txt 

        tato@finchat:~/lightning$ tail -f  /home/tato/lgt/test12/log.txt
        lightningd(4379): Hello world from v0.5.2-2016-11-21-752-gfc59d8e!
        lightningd(4379): Our ID: 02926ff80c40870a09d5e7f0ca20a482f44b9dc757c54c892747e480a8b42c8f83
        lightningd_gossip(4388): pid 4388, msgfd 13
        lightningd(4379): Failed to bind on 2 socket: Address already in use


    #================================
    # Run  localhost:42013 test13 


      #--------------------------------
      # Start lightningd  test13 

        $ $HOME/lightning/lightningd/lightningd  --port 42013 --lightning-dir /home/tato/lgt/test13 --log-file /home/tato/lgt/test13/log.txt & 


      #--------------------------------
      # tail 5 lines of logs: /home/tato/lgt/test13/log.txt 

        $ tail -5 /home/tato/lgt/test13/log.txt 

        tato@finchat:~/lightning$ tail -f  /home/tato/lgt/test13/log.txt
        lightningd(4956): Hello world from v0.5.2-2016-11-21-752-gfc59d8e!
        lightningd(4956): Our ID: 031b861c072f51dc89462b1f9d0ca85eb27c87dee643a1fdb9f589b620490ca85c
        lightningd_gossip(4965): pid 4965, msgfd 13
        lightningd(4956): Failed to bind on 2 socket: Address already in use


      #--------------------------------
      # Maked new address -->@return <addr>

        $ $HOME/lightning/daemon/lightning-cli  --lightning-dir /home/tato/lgt/test12 newaddr 

         { "address" : "2N8dgFvQjBvijeTS8uMMZjzZoWsvmP2sUmw" }

      #--------------------------------
      # Send BTC to test12 -->@return <txid>

        $ bitcoin-cli -testnet sendtoaddress  2N8dgFvQjBvijeTS8uMMZjzZoWsvmP2sUmw 0.003

         9888c3188a3749ad70811502bbd8336b502686674f1ae41df6d03d5b2816791a

      #--------------------------------
      # Retrieves the raw transaction -->@<rawtx>

        $ bitcoin-cli -testnet getrawtransaction  9888c3188a3749ad70811502bbd8336b502686674f1ae41df6d03d5b2816791a

         0200000001489671442eaf0bcef8538df0e41b882b3a1d309c79d27d41e1065f750a41ea46010000006a47304402203899628c08f217f264d043e8360b4e6c0d756e5d5b4d64f090ded865292d70bf02206a129b1223676e1a0ad38357b57a138757ff1deed9fbccfcb4992fcd59775e9b0121034413e43a0037dd7fbd71d0ed0444c8567a6992e86a90db4a13626f5310f60f7efeffffff02e09304000000000017a914a8c8f778cacb0d8b63cdee6e435308f64f70c4e2872cf08c00000000001976a91497190c7f45de4a0c9f725613264237107fbf147f88ac26fa1100

      #--------------------------------
      # Notifies `lightningd` that there are now funds available

        $ $HOME/lightning/daemon/lightning-cli --lightning-dir /home/tato/lgt/test12 addfunds 0200000001489671442eaf0bcef8538df0e41b882b3a1d309c79d27d41e1065f750a41ea46010000006a47304402203899628c08f217f264d043e8360b4e6c0d756e5d5b4d64f090ded865292d70bf02206a129b1223676e1a0ad38357b57a138757ff1deed9fbccfcb4992fcd59775e9b0121034413e43a0037dd7fbd71d0ed0444c8567a6992e86a90db4a13626f5310f60f7efeffffff02e09304000000000017a914a8c8f778cacb0d8b63cdee6e435308f64f70c4e2872cf08c00000000001976a91497190c7f45de4a0c9f725613264237107fbf147f88ac26fa1100

         { "outputs" : 1, "satoshis" : 300000 }

      #--------------------------------
      # The node ID by getinfo to test13 -->@return <node-id>

        $ $HOME/lightning/daemon/lightning-cli --lightning-dir /home/tato/lgt/test13 getinfo 

         { "id" : "031b861c072f51dc89462b1f9d0ca85eb27c87dee643a1fdb9f589b620490ca85c", "port" : 42013, "testnet" : true, "version" : "v0.5.2-2016-11-21-752-gfc59d8e", "blockheight" : 1178150 }

      #--------------------------------
      # Connect from test12 to test13

        $ $HOME/lightning/daemon/lightning-cli  --lightning-dir /home/tato/lgt/test12 connect localhost 42013 031b861c072f51dc89462b1f9d0ca85eb27c87dee643a1fdb9f589b620490ca85c

         { "id" : "031b861c072f51dc89462b1f9d0ca85eb27c87dee643a1fdb9f589b620490ca85c" }

      #--------------------------------
      # Use the connection to open a channel 

        $ $HOME/lightning/daemon/lightning-cli  --lightning-dir /home/tato/lgt/test12 fundchannel 031b861c072f51dc89462b1f9d0ca85eb27c87dee643a1fdb9f589b620490ca85c 20000

         {  }
         
      #--------------------------------
      # Check getpeers test12 and test13

        $ $HOME/lightning/daemon/lightning-cli  --lightning-dir /home/tato/lgt/test12 getpeers
        {
            "peers": [{
                "unique_id": 0,
                "state": "CHANNELD_AWAITING_LOCKIN",
                "netaddr": "127.0.0.1:42013",
                "peerid": "031b861c072f51dc89462b1f9d0ca85eb27c87dee643a1fdb9f589b620490ca85c",
                "owner": "lightningd_channel",
                "msatoshi_to_us": 20000000,
                "msatoshi_total": 20000000
            }]
        }

        $ $HOME/lightning/daemon/lightning-cli  --lightning-dir /home/tato/lgt/test13 getpeers
        {
            "peers": [{
                "unique_id": 0,
                "state": "CHANNELD_AWAITING_LOCKIN",
                "netaddr": "::ffff:127.0.0.1:48934",
                "peerid": "02926ff80c40870a09d5e7f0ca20a482f44b9dc757c54c892747e480a8b42c8f83",
                "owner": "lightningd_channel",
                "msatoshi_to_us": 0,
                "msatoshi_total": 20000000
            }]
        }

      #--------------------------------
      # Creates an invoice from test13

        $ $HOME/lightning/daemon/lightning-cli  --lightning-dir /home/tato/lgt/test13 invoice 1000 tx-01

         { "rhash" : "14d18c45c17e4e8a81228fb429b593314b0c89c6cb67131b500ee61bac866f81" 

      #--------------------------------
      # The node ID by getinfo to test13

        $ $HOME/lightning/daemon/lightning-cli  --lightning-dir /home/tato/lgt/test13 getinfo 
        {
            "id": "031b861c072f51dc89462b1f9d0ca85eb27c87dee643a1fdb9f589b620490ca85c",
            "port": 42013,
            "testnet": true,
            "version": "v0.5.2-2016-11-21-752-gfc59d8e",
            "blockheight": 1178151
        }
     
      #--------------------------------
      # The sender (test12) needs to compute a route to the recipient (test13)
     
        $ $HOME/lightning/daemon/lightning-cli  --lightning-dir /home/tato/lgt/test12 getroute 031b861c072f51dc89462b1f9d0ca85eb27c87dee643a1fdb9f589b620490ca85c 1000 1

      "Could not find a route"
      

route!

**Comment by toshirot** [comment]
This test is executed in the following flow
Is somewhere wrong?


    //Run lightningd host1 and host2

        //lightningd [<host1-port> <host1-dir> <host1-log> ...] //now sender
        //lightningd [<host2-port> <host2-dir> <host2-log> ...] //now recipient


    //Send BTC to host1(sender) by bitcoin-cli

        //1 lightning-cli [<sender-dir>] newaddr
              -->@return <addr>
              
        //2 bitcoin-cli sendtoaddress <addr> <amount>
              -->@return <txid>
              
        //3 bitcoin-cli getrawtransaction <txid>
              -->@return <rawtx>
              
        //4 lightning-cli [<sender-dir>] addfunds <rawtx>


    //Connect from host1 to host2

        //1 lightning-cli <recipient-dir> getinfo 
              -->@return { id: <recipient-node_id>, ...}
              
        //2 lightning-cli <sender-dir> connect <to-ip> <recipient-port> <recipient-node_id> 
              -->@return { id: <recipient-node_id> }
              
        //3 lightning-cli <sender-dir> fundchannel <recipient-node_id> <amount> 
              -->@return {  }


    //Make an invoice

        //1 lightning-cli invoice <msatoshi> <label>
              -->@return <rhash>


    //Sendpay from host1 to host2

        //1 lightning-cli <recipient-dir> getinfo
              -->@return { id: <recipient-node_id>, ...}
        //2 lightning-cli <sender-dir> getroute <recipient-id> <msatoshi> <riskfactor>
              -->@return $route object
        //3 lightning-cli sendpay $route <rhash>

**Comment by cdecker** [comment]
Ok, it appears you are not waiting for the channel's funding transaction to be confirmed. Notice that the channel only becomes usable after it is confirmed (2 confirmations are sufficient in regtest and testnet). On regtest run `./bitcoin-cli -regtest generate 2` to get 2 confirmations, on testnet just wait until 2 blocks are found.

As a side note you can check whether the channel is operational by using `lightning-cli getinfo`. It should display `state: CHANNELD_NORMAL` once it is ready. To see which nodes and channels are available in the network you can also use the `getnodes` and `getchannels` commands.
